### PR TITLE
Bc support content side

### PR DESCRIPTION
**Pull request by tomneeman151293**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39179

## Description
Adding support from content side to BC changes versions

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Tests/Marketplace/marketplace_services.py
Comment (guyfreund) [inline Tests/Marketplace/marketplace_services.py:2397]:
```suggestion
            changelog[rn_version]['breakingChanges'] = self.changelog_entry_contains_bc_version(predecessor_version,
                                                                                                rn_loose_version)
```
why do we update every entry? let's just update the relevant entries

File: Tests/Marketplace/marketplace_services.py
Comment (guyfreund) [inline Tests/Marketplace/marketplace_services.py:108]:
```suggestion
        self.breaking_changes_versions: List[LooseVersion] = []  # List of BC versions in the given pack. Initialized in load_user_metadata function
```

**Review by guyfreund — CHANGES_REQUESTED** [review]
A very elegant solution, Nice 👏 
Please test the relevant scenarios with an upload trigger.
After a successful upload run was made, please create a Self-Service 6.5 and configure it to use the created bucket (from the upload run)

File: Tests/Marketplace/release_notes_bc_calculator.py
Comment (Noy-Maimon) [inline Tests/Marketplace/release_notes_bc_calculator.py:68]:
```suggestion
        # Handle cases of two or more BC versions in entry.
```

**Review by Noy-Maimon — APPROVED** [review]

**Review by guyfreund — CHANGES_REQUESTED** [review]
Nice 👏  Few notes:
1. Please run the following upload scenarios:
    - A change in a RN file with no conf file (to see we did not break anything)
    - A change in a RN file with a conf file with breaking changes and notes
    - A change in a RN file with both changes and notes of an aggregated version
    - A change in a RN file which is aggregated with versions that not all of them contains the breaking changes text (to see that we handle the concatenation as expected)
2. If we've introduced a new file, I think it'll be better to make it more generic, so this file should capture all RN handling functions & classes expect for the `prepare_release_notes` function of the `Pack` object. We can call it `release_notes_handler` or something similar.

File: Tests/Marketplace/marketplace_services.py
Comment (guyfreund) [reply Tests/Marketplace/marketplace_services.py:960]: This is regarding line 960

Comment (tomneeman151293) [reply Tests/Marketplace/marketplace_services.py:960]: Fixed

**Comment by tomneeman151293** [comment]
Moved all the logic of the new added class inside MP file. also moved tests from the new calc test files to the mp tests

**Review by guyfreund — COMMENTED** [review]
looks good, waiting for the approval for when we have the runs attached

File: Tests/Marketplace/marketplace_services.py
Comment (dantavori) [inline Tests/Marketplace/marketplace_services.py:2540]:
```suggestion
            breaking_changes_versions (List[LooseVersion]): List of BC versions.
```

**Review by dantavori — APPROVED** [review]
looks great 👍

File: Tests/Marketplace/marketplace_services.py
Comment (Noy-Maimon) [inline Tests/Marketplace/marketplace_services.py:2527]:
```suggestion
            if file_data.get('breakingChanges'):
```
Redundant

**Comment by tomneeman151293** [comment]
@guyfreund 
Sending zip, containing checks of
new RN without BC
new RN with BC
new RN aggregated some BC versions some aren't
[bc_checks.zip](https://github.com/demisto/content/files/7015480/bc_checks.zip)

File: Tests/Marketplace/marketplace_services.py
Comment (Noy-Maimon) [inline Tests/Marketplace/marketplace_services.py:2513]: Worth adding a comment about the json files - that the new practice about BC would be adding that json file to the RN dir.

File: Tests/Marketplace/marketplace_services.py
Comment (Noy-Maimon) [inline Tests/Marketplace/marketplace_services.py:2418]:
```suggestion
                changelog[changelog_entry].pop('breakingChanges')
```

File: Tests/Marketplace/marketplace_services.py
Comment (Noy-Maimon) [inline Tests/Marketplace/marketplace_services.py:2415]:
```suggestion
                    changelog[changelog_entry].pop('breakingChangesNotes')
```

File: Tests/Marketplace/marketplace_services.py
Comment (Noy-Maimon) [reply Tests/Marketplace/marketplace_services.py:2418]: Are you sure this required? Why should the changelog have a 'breakingChanges' key if it didn't have any bc versions? (thats the `if` condition to this `else`)

File: Tests/Marketplace/marketplace_services.py
Comment (tomneeman151293) [reply Tests/Marketplace/marketplace_services.py:2415]: This will raise an error if breakingChangesNotes did not exist in the dict.

File: Tests/Marketplace/marketplace_services.py
Comment (tomneeman151293) [reply Tests/Marketplace/marketplace_services.py:2418]: If some version was BC but then someone decided to modify it to be a non-BC version

**Comment by tomneeman151293** [comment]
@guyfreund Added examples from runs.
@Noy-Maimon Fixed CR notes (or commented on other requests that weren't changed for its reason)

**Review by guyfreund — APPROVED** [review]

**Review by Noy-Maimon — APPROVED** [review]

**Comment by tomneeman151293** [comment]
This PR will be merged after the modifications done in 
https://github.com/demisto/demisto-sdk/pull/1579
for demisto-sdk to support conf files for RN will be merged.

**Comment by ShahafBenYakir** [comment]
@tomneeman151293 I think we mentioned we'll also add a an argument to SDK update release notes command that will create the json file automatically no?

**Comment by tomneeman151293** [comment]
@ShahafBenYakir This has a different PR and a different issue
https://github.com/demisto/etc/issues/39725

**Comment by ShahafBenYakir** [comment]
Yeah I know just wanted to make sure we are also adding that flag. Now I saw you have a PR.

**Pull request merged**
